### ISSUE TITLE
Request: Add Android port welcome screen to translation extras.

### ISSUE THREAD

**Issue by Graywolfe813** [CleverRaven/Cataclysm-DDA#21016]
We're getting feedback from users have some issues with basic controls on the Android port. May I PR our welcome screen for localization?

**Comment by kevingranade** [comment]
I don't mind you using our translation infrastructure but you need to
arrange to just inject the strings. I don't want to merge non-functional
partial copies of the Android port code.
Merging the port itself would also be fine of course.

**Comment by Graywolfe813** [comment]
My intent was to configure it for localization, and then PR it as an additional file to lang/extra/android, as was done with the options. That would be acceptable?

@a1studmuffin in regards to a merge of the port itself.

TL;DR; - There are issues, and I wouldn't merge it yet, but you could if you wanted to.

My personal opinion is that it's not really ready to go mainline. There is an issue involving Android libstdc++ implementations, for one. The only libstdc++ that functions properly is the gcc implementation, and the Android version of that is missing a swathe of functions from the std namespace added in C++11, primarily math functions. Every std::[whatever] call that's missing has required an ugly platform specific preprocessor definition that adds near-duplicate code. It works, but it's not satisfactory.

I'm also concerned about how the Android port for SDL functions in general. It runs two threads, one on the Dalvik VM, and one in pure native code, and makes a slew of jni calls back and forth between the two. It's not stable. There are EGL thread safety problems, and I've managed to completely break the event loop by switching in and out of the app rapidly. I'd like to switch the whole thing over to a native activity, which would remove the need for a second thread completely.

When it comes down to it, though, you could merge it any time you choose. It should be as simple as fetch master and merge, although it may be necessary to fetch the replace ref first.

**Comment by kevingranade** [comment]
Question answered, either merge the port into mainline, or arrange to inject the strings, but merging non-functional code is not ok.
